File: app/models/national_governing_body_stat.rb
# == Schema Information
#
# Table name: national_governing_body_stats
#
#  id                         :bigint           not null, primary key
#  assistant_referees_count   :integer          default(0)
#  community_teams_count      :integer          default(0)
#  competitive_teams_count    :integer          default(0)
#  developing_teams_count     :integer          default(0)
#  end_time                   :datetime
#  head_referees_count        :integer          default(0)
#  inactive_teams_count       :integer          default(0)
#  snitch_referees_count      :integer          default(0)
#  start                      :datetime
#  team_status_change_count   :integer          default(0)
#  total_referees_count       :integer          default(0)
#  total_teams_count          :integer          default(0)
#  uncertified_count          :integer          default(0)
#  university_teams_count     :integer          default(0)
#  youth_teams_count          :integer          default(0)
#  created_at                 :datetime         not null
#  updated_at                 :datetime         not null
#  national_governing_body_id :bigint
#
# Indexes
#
#  ngb_stats_on_ngb_id  (national_governing_body_id)
#
# Foreign Keys
#
#  fk_rails_...  (national_governing_body_id => national_governing_bodies.id)
#

class NationalGoverningBodyStat < ApplicationRecord
  belongs_to :national_governing_body

  def self.by_month(month)
    where('EXTRACT(month FROM end_time::date)::integer = ?', month)
  end

  def update_team_counts
    Services::UpdateTeamCounts.new(self).perform
  end
end

File: dotnet/ManagementHub.Models/NationalGoverningBodyStat.cs
using System;
using ManagementHub.Models.Abstraction;

namespace ManagementHub.Models
{
	public partial class NationalGoverningBodyStat : IIdentifiable
	{
		public long Id { get; set; }
		public long? NationalGoverningBodyId { get; set; }
		public int? TotalRefereesCount { get; set; }
		public int? HeadRefereesCount { get; set; }
		public int? AssistantRefereesCount { get; set; }
		public int? SnitchRefereesCount { get; set; }
		public int? CompetitiveTeamsCount { get; set; }
		public int? DevelopingTeamsCount { get; set; }
		public int? InactiveTeamsCount { get; set; }
		public int? YouthTeamsCount { get; set; }
		public int? UniversityTeamsCount { get; set; }
		public int? CommunityTeamsCount { get; set; }
		public int? TeamStatusChangeCount { get; set; }
		public DateTime CreatedAt { get; set; }
		public DateTime UpdatedAt { get; set; }
		public int? TotalTeamsCount { get; set; }
		public int? UncertifiedCount { get; set; }
		public DateTime? Start { get; set; }
		public DateTime? EndTime { get; set; }
		public int? ScorekeeperRefereesCount { get; set; }

		public virtual NationalGoverningBody? NationalGoverningBody { get; set; }
	}
}
